### PR TITLE
badges: Add icon to menu and reformat dialog

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Minor tweaks for dialog usage.

1. Addition of an icon to the entity context menu

| Before | After |
| --- | --- |
| image | image |

2. Reformatting of the dialog, hopefully to simplify and make it easier to read. Removed the descriptive headings as I think it may be obvious, _but_ welcome feedback/adjustments for other opinions.

| Before | After |
| --- | --- |
| image |  image |

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/badges/src/components/EntityBadgesDialog.tsx [hunk 1]
```diff
@@ -43,19 +43,10 @@ type Props = {
   entity: Entity;
 };
 
-const useStyles = makeStyles({
-  codeBlock: {
-    '& code': {
-      whiteSpace: 'pre-wrap',
-    },
-  },
-});
```
Comment (adamdmharvey) [inline plugins/badges/src/components/EntityBadgesDialog.tsx:52]: Note removed all of the useStyles/classes formatting and such as when I did, there did not look to be any visual difference, so I believe the `<CodeSnippet>` component already drove the appropriate formatting.

File: plugins/badges/src/components/EntityBadgesDialog.tsx [hunk 2]
```diff
@@ -87,6 +71,10 @@ export const EntityBadgesDialog = ({ open, onClose, entity }: Props) => {
       <DialogTitle>Entity Badges</DialogTitle>
 
       <DialogContent>
+        <Typography>
```
Comment (freben) [inline plugins/badges/src/components/EntityBadgesDialog.tsx:74]: `DialogContentText`.

**Review by freben — APPROVED** [review]
Just one comment, but pre-approving so it's fine to merge after you switch that component

File: plugins/badges/src/components/EntityBadgesDialog.tsx (hunk 2 above)
Comment (adamdmharvey) [reply plugins/badges/src/components/EntityBadgesDialog.tsx:74]: Will do.  It's interesting, as there's actually a look in here so it ends up being...

* DialogContent
  * Typograhy
  * DialogContentText (each badge)
  * DialogContentText (each badge)
  * DialogContentText (each badge)

But updated that first one per your feedback and tested locally.

**Pull request merged**